### ISSUE TITLE
Make a unified result printer for Python tools

### ISSUE THREAD

**Issue by palmtenor** [iovisor/bcc#1110]
We now have a lot of Python tools. Each of them had to implement their own output logic, no matter it's polling to output real-time events, aggregating results after profiling, our making histogram stats. And those logic have a lot in common.

I think we should have a unified result printer library to be shared across the standalone Python CLI tools. This way:
- When making new tools, no need to do duplicated work for formatting and printing output.
- We could easily improve the library and add new output functionalities across all tools, like options to control what to print (especially for stack symbols), add different formats like `csv`, etc. so the profiling result can be easily passed to other tools.

**Comment by goldshtn** [comment]
Agreed, although a lot of tools have slightly varying opinions on what should be printed and when. At the bare minimum, a decent table printing API would be really useful for many tools. Same goes for exporting to different formats (when it comes to tables).

**Comment by palmtenor** [comment]
I guess we could start with the "polling events" ones, like `trace.py`, `*snoop.py` etc.? It should be pretty reasonable to get them to unified process information and stack output options.

**Comment by brendangregg** [comment]
Maybe, but bear in mind this is a tracing tool, and I think people have limited appetite to be learning an API. So keep it as small as possible.

I think SystemTap got this way wrong, as an example, and has way too many things in their tapset API. It puts off newbies.

**Comment by martinvonwittich** [comment]
> We could easily improve the library and add new output functionalities across all tools, like options to control what to print (especially for stack symbols), add different formats like `csv`, etc. so the profiling result can be easily passed to other tools.

Different output formats like CSV or JSON would be very much appreciated. I'm thinking about post-processing the output of tools like `exitsnoop-bpfcc`, but the current space-separated output format makes that rather tricky:

```
martin.iserv.dev ~ # cat foo\ bar 
#!/bin/sh
sleep 1
```

```
martin.iserv.dev ~ # exitsnoop-bpfcc -t
[...]
14:51:29.382 sleep            575686 575685 575686 1.00    0
14:51:29.383 foo bar          575685 572894 575685 1.00    0
[...]
```
